app/controllers/v0/profile/connected_applications_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module V0
+  module Profile
+    class ConnectedApplicationsController < ApplicationController
+      def index
+        render json: { data: grants_by_app.values }
+      end
+
+      def destroy
+        app = grants_by_app(false)[connected_accounts_params[:id]]
+        app[:attributes][:grants].each do |grant|
+          delete_response = okta_service.delete_grant(@current_user.uuid, grant[:id])
+          unless delete_response.success?
+            log_message_to_sentry("Error deleting grant #{connected_accounts_params[:id]}", :error,
+                                  body: delete_response.body)
+            raise 'Unable to delete grant'
+          end
+        end
+
+        head :no_content
+      end
+
+      private
+
+      def okta_service
+        @okta_service ||= Okta::Service.new
+      end
+
+      def get_logo(app_id)
+        app_response = okta_service.app(app_id)
+        if app_response.success?
+          app_response.body['_links']['logo'].last['href']
+        else
+          log_message_to_sentry('Error fetching app', :error,
+                                body: app_response.body)
+          raise 'Unable to fetch app'
+        end
+      end
+
+      def apps_from_grants(grants, with_logos = true)
+        apps = {}
+        grants.body.each do |grant|
+          links = grant['_links']
+          app_href = links['app']['href']
+          app_id = app_href.split('/').last
+
+          unless apps[app_id]
+            apps[app_id] = {
```

Comment:
Any thought into making okta grants and applications in to a Redis module? (see `app/models/user.rb` specifically the `emis attributes section`). It'd probably be good to cache applications across users to save on the logo request. It also means you could use the regular `ActiveModel::Serializers` other end points use, so serialization logic isn't in the controller.

Reply:
Given we know now we have to look up the user -> look up grants -> look up applications. I think we'll need to follow the same model we do for lazy loading attributes and saving to redis we do for other services in `api/models/user.rb`

Reply:
Thought actually we can probably skip for now and ticket this up as an enhancement for next sprint.